superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -339,16 +352,17 @@ function dbReducer(
         };
       }
 
+      // convert query to a string and store in query_input
+      query = action.payload?.parameters?.query || {};
+      query_input = Object.entries(query)
+        .map(([key, value]) => `${key}=${value}`)
+        .join('&');
```

Comment:
I think elsewhere we do this conversion at the input level, don't we?

Reply:
What do you mean with "at the input level"?

Reply:
I don't think we're doing this anywhere else.

With this PR we convert query to a string and store it in `query_input` when we fetch the database. Changes to `query_input` are passed to `query`, but this is might not survive a roundtrip (`a=b&` => `{a: 'b'}` => `a=b`), so we keep `query_input` always up to date and use it to populate the form input.

Reply:
I was thinking of what you did here, where you stored the schemas in state as an array, and then converted to a string on the input: https://github.com/betodealmeida/incubator-superset/blob/e768b3d20a554eb3da2b1f1f6910477fecaab9ee/superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx#L377
But maybe that won't work here.. I think it's ok to leave this as is.. I was thinking consistency and keeping the state as simple as possible.

Reply:
It works for the array because the roundtrip preserves the text in the form:

```
a => ['a'] => a
a, => ['a', ''] => a,
```

But for query we need to store the intermediary value for cases like this:

```
a=1&b => {a: 1, b: ''} => a=1&b=
```